Build failure on macOS and Intel with MAPL3

I think some odd behavior with Intel + macOS has changed recently. Code that is pretty dang old has now stopped working. To wit:
```
[ 15%] Linking Fortran shared library ../lib/libMAPL.profiler.dylib
... delete a lot of warnings ...
Undefined symbols for architecture x86_64:
  "_mallinfo", referenced from:
      _mapl_mallocgauge_mp_get_measurement_ in MallocGauge.F90.o
ld: symbol(s) not found for architecture x86_64
```

Thing is `MallocGauge.F90` hasn't changed...ever? So this must be an Intel issue? Or something? 

I'm going to assign @tclune as this is his code. I suppose a simple "if darwin and intel don't do this" is all that's needed, but I'm not exactly sure how to correctly do that here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,9 +9,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Add define for `-Dsys${CMAKE_SYSTEM_NAME}` to fix build issue with macOS and Intel (#1695)
+
 ### Added
 
 ### Changed
+
 - Error codes in `shared/MAPL_Error_Handling.F90` are now consistent with `_FAILURE = 1` in `include/MAPL_ErrLog.h`
 
 ### Removed

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -158,6 +158,8 @@ else ()
   add_subdirectory (pflogger_stub)
 endif ()
 
+add_definitions(-Dsys${CMAKE_SYSTEM_NAME})
+
 # Special case - MAPL_cfio is built twice with two different precisions.
 add_subdirectory (MAPL_cfio MAPL_cfio_r4)
 add_subdirectory (MAPL_cfio MAPL_cfio_r8)

